### DISCUSSION TITLE
Cannot identify any devices

### DISCUSSION THREAD

**Discussion by Anonymouse83** [fsaris/EspHome-AwoX-BLE-mesh-hub#87] [General]
My apologies if this is an ignorant question.
I am trying to connect to my EGLO bulb using this library, but looking at the code sample, it seems I need the MAC address, which I cannot identify.
Ideally, I would like to be able to search for attached devices and then report back on their status/ control them, rather than knowing the MAC address of each device (which I cannot seem to identify).
Any help would be most appreciated.

**Comment by fsaris** [comment]
That's actually what the component does.

The options related to Mac adres etc it only for finetunen the network. But in general not needed.

You will need your mesh credentials, easiest way to find/get these is to use this form (https://fsaris.github.io/EspHome-AwoX-BLE-mesh-hub/awoxh-mesh-credentials-tool/ ) to read them from your AwoX Cloud account

Then the component will try to find all devices

**Reply by Anonymouse83** [reply]
Many thanks for the reply.
I was basing my code on the following:-
`
import awoxmeshlight
mylight = awoxmeshlight.AwoxMeshLight ("A4:C1:38:78:11:33")
mylight.connect() # Unless specified, the default name and password are used
mylight.setMesh("new_mesh_name", "new_mesh_password","new_mesh_key")
mylight.disconnect()
`
This indicates that the MAC address is needed before anything else - but I couldn't see how to connect to the mesh to obtain the devices.
Any help would be most appreciated.

**Reply by fsaris** [reply]
Just use the example yaml as starting point https://github.com/fsaris/EspHome-AwoX-BLE-mesh-hub?tab=readme-ov-file#yaml-options

No need to write your own code

**Reply by Anonymouse83** [reply]
Thank you for your help.
I looked at the yaml code, but it wasn't clear as to how that will help.
I have written (and am extending) my own application - which is doing quite a lot of different things - and the idea is not to use another app, but to use my Python code to connect to the device(s) and to read the status - and control them from my Python app.
Therefore, I need to be able to connect to the mesh, see the devices, and then - if appropriate read the status/ control the device(s).
I thought your python-awox-mesh-light library was supposed to do that, but I cannot see how.
If there is a more suitable library for my needs, I would be grateful for guidance - or how to connect to the mesh for this control!
Many thanks,

**Reply by fsaris** [reply]
okay, this is no python code but C++ especialy for running with ESPHome.

The https://github.com/fsaris/home-assistant-awox component uses python but due to some issues I switched to ESPHome https://github.com/fsaris/home-assistant-awox/issues/110

both solutions (esphome and python) scan for Bluetooth devices and try to find the fist light to connect to based on the mac address prefix. After connecting the light and the rest of the lights can be controlled through that connection

**Reply by Anonymouse83** [reply]
My apologies, but I am now rather confused.
There is a python library that is supposed to access the Awox devices.
I was unable to get that to access my devices.
My ideal would be to be able to scan for available devices, then connect to the devices, and then read/ control remotely.
You mentioned that I don't need the MAC address, but I could not see how to scan for the devices - nor how to connect to an unknown device.
If you are suggesting there is a different Home Automation library that would be required/ helpful, are you able to recommend that?
Many thanks for your help - I am very new to this,and maybe overreaching my capabilities!

**Reply by fsaris** [reply]
See https://github.com/fsaris/home-assistant-awox/blob/e6e5378eed717ee40183ad34a6109d484f5052fe/custom_components/awox/scanner.py#L61
